### PR TITLE
[sc-25976] Add `database_defaults` to Metabase crawler config

### PR DESCRIPTION
**Pull request by usefulalgorithm**
### 🤔 Why?

Metabase does not know a SQL query's default schema on Snowflake, RedShift and BigQuery. This causes our lineage parser unable to locate the correct upstream dataset, and in some instances the lineage parser to straight up fail.

Since Metabase API does not provide us with the default schema, best we can do is to specify this in the configs.

### 🤓 What?

- Add config for `database_defaults`, which contains `default_schema`
- Use a database's `default_schema` when parsing database info from metabase api

### 🧪 Tested?

- Unit test
- Tested locally

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/metabase/extractor.py [hunk 1]
```diff
@@ -210,21 +211,36 @@ def _parse_database(self, database: Dict) -> None:
...
             # not able to get connection details, possibly due to lack of Admin permission
             return
 
         if platform == DataPlatform.SNOWFLAKE:
             self._databases[database_id] = DatabaseInfo(
-                platform, details.get("db"), None, details.get("account")
+                platform,
+                details.get("db"),
+                default_schema,
+                details.get("account"),
```
Comment (elic-eon) [inline metaphor/metabase/extractor.py:232]: Didn't Metabase provide any schema information? Then, do we need to specify the default schema for every (Metabase) database?

Comment (elic-eon) [reply metaphor/metabase/extractor.py:232]: what if the query is using the full table name, `SELECT * from db.schema.table`. Can we get the schema in the lineage parser?

Comment (elic-eon) [reply metaphor/metabase/extractor.py:232]: Metabase may know which RS schema is queried against but we can tell from the API. There is another possibility, there is some default schema in the RS system. When we use this SQL, `SELECT * from TABLE`, there is a default schema.

Comment (mars-lan) [reply metaphor/metabase/extractor.py:232]: The default schema is tied to the account used so it's not possible to tell that from Metabase side.

File: metaphor/metabase/README.md [hunk 2]
```diff
@@ -25,6 +25,16 @@ password: <password>
 
 ### Optional Configurations
 
+#### Database Defaults
+
+Metabase parses the SQL queries executed on Redshift, Snowflake and BigQuery, but it is unaware of a query's default schema during its execution. Metaphor's in-app SQL lineage parser will not be able to locate the correct dataset if the query's schema is not provided. To set the SQL platform's default schema, set the `database_defaults` configuration:
```
Comment (mars-lan) [inline metaphor/metabase/README.md:30]:
```suggestion
Metabase's API does not provide information on the default schema used to execute [native queries](https://www.metabase.com/glossary/native_query). This makes it difficult to parse the lineage precisely. When this happens, use `database_defaults` to manually set the [database](https://www.metabase.com/docs/latest/databases/start)'s default schema:
```

File: metaphor/metabase/README.md [hunk 3]
```diff
@@ -25,6 +25,16 @@ password: <password>
...
+
+```yaml
+database_defaults:
+    - id: <id of the database in Metabase>
+      default_schema: <default schema in the SQL platform>
```
Comment (mars-lan) [inline metaphor/metabase/README.md:35]:
```suggestion
      default_schema: <default schema for the database>
```

File: metaphor/metabase/README.md (hunk 2 above)
Comment (usefulalgorithm) [reply metaphor/metabase/README.md]: noice

File: metaphor/metabase/extractor.py (hunk 1 above)
Comment (usefulalgorithm) [reply metaphor/metabase/extractor.py:232]: > what if the query is using the full table name, SELECT * from db.schema.table. Can we get the schema in the lineage parser?

then it is ok

> Metabase may know which RS schema is queried against but we can tell from the API. ...

The tricky part is that MB doesn't know about RS schema, instead it only knows RS database.

**Pull request merged**